### DISCUSSION TITLE
Add options to enable/disable mob adjustments

### DISCUSSION THREAD

**Discussion by NickvdMeij** [P3pp3rF1y/SophisticatedBackpacks#858] [General]
In some cases, you might not want mobs to have backpacks, or have mobs altered/added while only having the backpack functionality for the players. Currently, there is no way of turning this behaviour off, leading to annoying edge cases that cannot be fixed.

Ideally there would be at least a few settings like:

- Enable custom mobs
- Enable backpacks for mobs

Furthermore, it is for the end user extremely annoying that this is not documentated anywhere. I already saw a few reddit posts regarding this and people wondering where the "OP zombie with music" comes from, only to found out its this mod that adds it.

Last but not least, regarding the OP zombie that you guys add in the mod, please nerf it a bit. I can tell you from experience that having that dude spawn near spawn, while you have just spawned into a world leads to a non-stop insta death for the player that he can't do anything about. What is the reason you have a zombie so many buffs and so much health? He is almost as tough as a boss, which makes no sense at all

**Comment by P3pp3rF1y** [comment]
You can control both of these in config, there's a chance and if you set it to 0 no mobs will spawn with backpacks, there's also list of entities and related loot tables that can spawn with backpacks (I would just say that backpack may not render properly on some modded entities so it's really up to whoever configures this to test). 
And regarding way too OP mobs I tuned down a few effects, but if really op one spawned close to spawn that was the most unlucky thing to happen. By default 1 in 100 mobs can spawn with backpack but because not all entities can spawn with it (like spiders) it's more like 1 in 200 and that's for the base backpack where the mob will only have minor effect applied and just a bit elevated health. For one that has diamond or netherite backpack the probability is more like 1 in 20,000 or 1 in 120,000

**Reply by NickvdMeij** [reply]
Awesome, the option is really helpful thanks!
In regards to the change, is it perhaps an idea to implement it that the diamond/netherite mobs perhaps don't spawn before you at least have the diamond/netherrite achievement? That way it would prevent them from spawning when the player is really not geared up, while still having them spawn when the player has viable gear to deal with them. Think this would be a more fun approach.
In the end, even a 1 in 120.000 chance will happen eventually with the more players and playthrough over time. As you can see from my experience, the 1 in 20.000 already happened (cause mine was the diamond backpack one). The most frustrating for me was that even though I could run from him, attacking was useless due to his armor and my stone axe. There was really no viable way to kill him, due to his amount of damage again no armor and my inability to kill him due to my rubbish tools. Having it achievement locked will ensure the player at least has some iron gear (hopefully, i mean you are mining diamonds after all), and therefor decrease the risk of being gamestopped by having an unlucky diamond/netherrite mob spawn.

**Reply by Alejo8988** [reply]
> You can control both of these in config...

I didn't really found a way to do so, is there really one?

**Reply by P3pp3rF1y** [reply]
@Alejo8988 there's a whole section with bunch of settings in config just search for `entityBackpackAdditions`
